Fix route_to issue

parisiam opened the pull request:
Fixes #267

- Replaced `route_to()` by `url_to()` in the views, when is was used alone.
- When `route_to()` was used with `site_url()`, as in `site_url(route_to(...))`, I've left it for the sake of diversity.
- Changed `route_to()` to `url_to()` also in `tests/Authentication/MagicLinkTest.php` (please check carefully for this one as I'm not too sure about it).
> First PR here, sorry in advance if I missed something. Remarks/advice welcome.

kenjis commented on tests/Authentication/MagicLinkTest.php:
```diff
@@ -37,29 +37,29 @@ protected function setUp(): void
     public function testCanSeeMagicLinkForm(): void
     {
-        $result = $this->get(route_to('magic-link'));
+        $result = $this->get(url_to('magic-link'));
```
Revert all the changes in this file. `$this->get()`, `$this->post()`, `$result->assertRedirectTo()` does not take absolute URL.

See
- https://codeigniter4.github.io/CodeIgniter4/testing/feature.html#requesting-a-page
- https://codeigniter4.github.io/CodeIgniter4/testing/response.html#assertredirectto

MGatner commented:
Thanks @parisiam! Glad to have your contribution. Please revert the test case changes as kenjis noted. 

> When `route_to()` was used with `site_url()` ... I've left it for the sake of diversity.

I would be in favor of consistency here, open to other opinions though.

kenjis commented:
I think we don't need to use  `site_url(route_to(...))`.
We use it just because we did not know `url_to()` works like that.

parisiam commented:
Thank you both for your comments and your help !! Really appreciate it.
- I've reverted the changes in MagicLinkTest.php
- I've replaced site_url(route_to(...)) by url_to(...) in email_2fa_show.php and email_2fa_verify.php
Don't hesitate to let me know if some further corrections are needed.

kenjis reviewed (approved):
Thank you.
LGTM!

MGatner reviewed (approved):
Very nice work! Thank you @parisiam
